plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/IcebergGlueCatalogModule.java

```diff
@@ -46,7 +47,8 @@
     protected void setup(Binder binder)
     {
         configBinder(binder).bindConfig(GlueHiveMetastoreConfig.class);
-        configBinder(binder).bindConfig(IcebergGlueCatalogConfig.class);
+        IcebergGlueCatalogConfig config = buildConfigObject(IcebergGlueCatalogConfig.class);
```

Comment:
`buildConfigObject()` is intended for conditional configuration of the injector. In this case, we don't need it at binding time, so it's cleaner to use a provider method, just like any other object that we need manually construct based on injected things:
```java
@Provides
@Singleton
@ForGlueHiveMetastore
public static RequestHandler2 createRequestHandler(IcebergGlueCatalogConfig config)
{
    return new SkipArchiveRequestHandler(config.isSkipArchive());
}
```
While using `buildConfigObject()` works fine, it feels like a "trick" or "cheat" that doesn't generalize to other situations. If we needed anything else from the injector to construct the instance, it wouldn't work.